### DISCUSSION TITLE
Windows mingw 11

### DISCUSSION THREAD

**Discussion by jeremy-rifkin** [actions/virtual-environments#5291] [Ideas]
I am trying to setup CI for C++ 20 on windows with gcc/mingw. The windows server environment comes with mingw-8 but newer is needed for C++20. `choco upgrade mingw` takes 5 minutes to run. Would it be possible to get a windows image that either has mingw-11 installed already or for the choco package to be cached so this step runs faster?

Edit: On a related not, for some reason I can compile but not run programs after the upgrade, they immediately exit with code 3221225785 and I am not sure of the dependency issue.

**Comment by miketimofeev** [comment]
Hey @jeremy-rifkin!
Have you tried using https://github.com/msys2/setup-msys2 action? It's a recommended way to install MinGW on windows environments. We can not update the MinGW version easily as it will break many other customers.

**Reply by jeremy-rifkin** [reply]
That'll work, thanks!
